Export Type Registry for Reporting [POC/exploration] (#155081)
elastic/kibana: x-pack/plugins/reporting_export_types/README.md

## Summary

There are three constructs that interact: the `ReportingPlugin`, the `ExportTypesPlugin` and the `ExportTypesRegistry`.

`ReportingExportTypesPlugin` on setup() creates the  `export_type_definitions`. The export type definitions can be one of the following:
- CSVExportTypeDefinition
- PNGExportTypeDefinition
- PDFExportTypeDefinition

> The `ReportingPlugin`  calls the `ReportingExportTypesPlugin` and creates an instance of the `ExportTypeRegistry`. These export types use the `ReportingPlugin`'s registerExportType() within the `ExportTypeRegistry`. The `ReportingPlugin` uses the getExportType() method to use the registered export types for exporting.

Reviewer: ```suggestion
The `ReportingPlugin` creates an instance of the `ExportTypeRegistry` and exposes its `registerExportType()` to other plugins, which `ReportingExportTypesPlugin` uses to register all of the export types. The `ReportingPlugin` uses the getExportType() method to use the registered export types for exporting.
```
